system/src/commonMain/kotlin/com/splendo/kaluga/system/network/state/NetworkStateRepo.kt

```diff
@@ -0,0 +1,124 @@
+            is Network.Unknown.WithoutLastNetwork -> NetworkState.Unknown(
+                Network.Unknown.WithoutLastNetwork(network.reason)
+            )
+            is Network.Unknown.WithLastNetwork -> NetworkState.Unknown(
+                Network.Unknown.WithLastNetwork(network.lastKnownNetwork, network.reason)
+            )
+            is Network.Known.Cellular -> NetworkState.Unknown(
+                Network.Unknown.WithoutLastNetwork(Network.Unknown.Reason.NOT_CLEAR)
+            )
+            is Network.Known.Wifi -> NetworkState.Unknown(
+                Network.Unknown.WithoutLastNetwork(Network.Unknown.Reason.NOT_CLEAR)
+            )
+            Network.Known.Absent -> NetworkState.Unknown(
+                Network.Unknown.WithoutLastNetwork(Network.Unknown.Reason.NOT_CLEAR)
+            )
+        }
+    }
+
+    internal fun onNetworkStateChange(network: Network) {
+        runBlocking {
+            takeAndChangeState { state: NetworkState ->
+                when (state) {
+                    is NetworkState.Available -> {
+                        when (network) {
+                            is Network.Unknown.WithoutLastNetwork -> {
+                                { state.unknownWithoutLastNetwork(network.reason) }
+                            }
+                            is Network.Unknown.WithLastNetwork -> {
+                                { state.unknownWithLastNetwork(network.lastKnownNetwork, network.reason) }
+                            }
+                            is Network.Known.Cellular -> state.availableWithCellular
+                            is Network.Known.Wifi -> state.availableWithWifi
+                            Network.Known.Absent -> state.unavailable
+                        }
+                    }
+                    is NetworkState.Unavailable -> {
+                        when (network) {
+                            is Network.Unknown.WithoutLastNetwork -> state.remain()
+                            is Network.Unknown.WithLastNetwork -> state.remain()
+                            is Network.Known.Cellular -> state.availableWithCellular
+                            is Network.Known.Wifi -> state.availableWithWifi
+                            Network.Known.Absent -> state.remain()
+                        }
+                    }
+                    is NetworkState.Unknown -> {
+                        when (network) {
+                            is Network.Unknown.WithoutLastNetwork -> state.remain()
+                            is Network.Unknown.WithLastNetwork -> state.remain()
+                            is Network.Known.Cellular -> state.availableWithCellular
+                            is Network.Known.Wifi -> state.availableWithWifi
+                            Network.Known.Absent -> state.unavailable
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+fun Flow<NetworkState>.network(): Flow<Network> {
```

Comment:
I would add one more util function `online` which emits a boolean only when you go online / offline (and initial state).

Then update the README with it.

Reply:
To be clear, where unknown = not online, so false.

Reply:
154cc3c